### PR TITLE
bug(rhineng-9037): Handle embedded percents and colons values

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-9037](https://issues.redhat.com/browse/RHINENG-9037).
* escape values with percents and colons to avoid unintended SQL interaction.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — CHANGES_REQUESTED** [review]
The changes look good, but it could use a test to validate it

File: tests/test_api_hosts_get.py [hunk 1]
```diff
@@ -1184,22 +1184,28 @@ def test_query_by_id_sparse_fields(db_create_multiple_hosts, api_get):
...
-        host_list_or_id=created_hosts[0].id, query="?fields[system_profile]=os_kernel_version,owner_id"
-    )
+    queries = [
+        "?fields[system_profile]=os_kernel_version,owner_id",
+        "filter[system_profile][bios_vendor]=%3E/%3AX%26x5wVZCj%25mC",
+    ]
+    for query in queries:
+        with subtests.test():
+            url = build_hosts_url(
+                host_list_or_id=created_hosts[0].id, query="?fields[system_profile]=os_kernel_version,owner_id"
```
Comment (vkrizan) [inline tests/test_api_hosts_get.py:1194]: Shouldn't the `query` argument take the `query` variable here?

Comment (chambridge) [reply tests/test_api_hosts_get.py]: 🤦 I've fixed the test case now.

**Review by kruai — APPROVED** [review]

File: api/filtering/db_custom_filters.py [hunk 2]
```diff
@@ -203,6 +212,7 @@ def build_single_text_filter(filter_param: dict) -> str:
         logger.debug(f"generating filter: field: {field_name}, type: {field_filter}, field_input: {field_input}")
 
         jsonb_path, pg_op, value = _convert_dict_to_text_filter_and_value(filter_param, field_filter == "array")
+        value = escape_sql_from_string_val(value)
```
Comment (vkrizan) [inline api/filtering/db_custom_filters.py:215]: Where is the value properly escaped so there are no more "other" SQL interactions?

**Pull request merged**